.spread-prepare.sh

```diff
@@ -0,0 +1,226 @@
+#!/bin/sh
+# This script is started by spread to prepare the execution environment
+set -x
+set -u
+set -e
+
+# Sanity check, are we in the top-level directory of the tree?
+test -f configure.ac || ( echo 'this script must be executed from the top-level of the tree' && exit 1)
+
+# Record where the top level directory is
+top_dir=$(pwd)
+
+# Record the current distribution release data to know what to do
+release_ID="$( . /etc/os-release && echo "${ID:-linux}" )"
+release_VERSION_ID="$( . /etc/os-release && echo "${VERSION_ID:-}" )"
+
+
+# Create source distribution tarball and place it in the top-level directory.
+create_dist_tarball() {
+    # Load the version number from a dedicated file
+    local pkg_version=
```

Comment:
I wonder if we could combine this and the following line to `local pkg_version=$(cat "$top_dir/VERSION)`. Using local in this way should be fine with dash https://wiki.ubuntu.com/DashAsBinSh#local - interessting enough `local` is not posix compliant, but I think we can ignore this :)